File: 08-1-2-State-Solution/States/HappyState.cs
namespace State.States
{
    internal sealed class HappyState : PartnerStateBase
    {
        public HappyState(Partner partner)
            : base(partner)
        {
        }

        public override void Talk()
        {
            Print("I am so happy!");
        }

        public override void Kiss()
        {
            Print("Thanks! <3 <3");
        }

        public override void Insult()
        {
            Print("Why would you say something like that? :-(");
            SetState(new AngryState(Partner));
        }
    }
}

File: 09-DesignPatterns-Java/Solution/State/src/States/HappyState.java
package States;

public final class HappyState extends PartnerStateBase {

    public HappyState(Partner.Partner partner) {
        super(partner);
    }

    @Override
    public void talk() {
        print("I am so happy!");
    }

    @Override
    public void kiss() {
        print("Thanks! <3 <3");
    }

    @Override
    public void insult() {
        print("Why would you say something like that? :-(");
        setState(new AngryState(_partner));
    }
}
